pallets/amm/src/lib.rs

```diff
@@ -155,107 +155,123 @@ pub mod pallet {
                 false => (liquidity_amounts.0, liquidity_amounts.1),
             };
 
-            if Pools::<T, I>::contains_key(base_asset, quote_asset) {
-                let pool_liquidity_amount: PoolLiquidityAmount =
-                    Self::pools(base_asset, quote_asset);
+            Pools::<T, I>::try_mutate(
+                &base_asset,
+                &quote_asset,
+                |pool_liquidity_amount| -> DispatchResultWithPostInfo {
+                    if let Some(liquidity_amount) = pool_liquidity_amount {
```

Comment:
let's avoid nested if scope here, maybe you can consider handling `None` case first and then quick return

Reply:
@GopherJ I have to handle both cases here a) When there is already a pool with same currency set b) When there is no pool

cc @ETeissonniere